modules/caddyhttp/proxyprotocol/listenerwrapper.go

```diff
@@ -39,31 +38,29 @@ type ListenerWrapper struct {
 	// allow/require PROXY headers from.
 	Allow []string `json:"allow,omitempty"`
 
-	rules []proxyprotocol.Rule
+	policies []goproxy.PolicyFunc
```

Comment:
Why store this in a slice when there can be one func at once? proxyproto takes a slice of IP address and return a function. There is no need to store it in a slice.

Reply:
It's my attempt at future-proofing. proxyproto has a selection of [policies](https://pkg.go.dev/github.com/pires/go-proxyproto#PolicyFunc). We can also build more as long as they conform to the `PolicyFunc` type. My thinking is we could expand the implementation later to unionize policies.